geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewaySenderCommand.java

```diff
@@ -149,9 +155,55 @@ public ResultModel createGatewaySender(
     ResultModel resultModel = ResultModel.createMemberStatusResult(gatewaySenderCreateResults);
     resultModel.setConfigObject(configuration);
 
+    if (!waitForGatewaySenderMBeanCreation(id, membersToCreateGatewaySenderOn)) {
+      resultModel.addInfo()
+          .addLine("Did not complete waiting for GatewaySenderMBean proxy creation");
+    }
+
     return resultModel;
   }
 
+  /*
+   * Wait for up tp 3 seconds for the proxy MBeans to be created.
+   */
+  @VisibleForTesting
+  boolean waitForGatewaySenderMBeanCreation(String id,
+      Set<DistributedMember> membersToCreateGatewaySenderOn) {
+    DistributedSystemMXBean dsMXBean = getManagementService().getDistributedSystemMXBean();
+    long startWaitTime = System.currentTimeMillis();
+
+    do {
+      try {
+        if (membersToCreateGatewaySenderOn.stream()
+            .allMatch(m -> gatewaySenderBeanExists(dsMXBean, m.getName(), id))) {
+          return true;
+        }
+
+        Thread.sleep(100);
+      } catch (InterruptedException e) {
+        // ignored
+      }
+
+    } while (System.currentTimeMillis() - startWaitTime < 3000);
```

Comment:
how was the time of 3 seconds chosen?  Is it possible that on a very large cluster this could take longer?  at the very least this should be a constant at the top of the class (or configurable?) not hardcoded deep in the code...

Reply:
The value was arrived at after running the test through `repeatTest` a bunch. I've now made it a constant and bumped it to 10 seconds. I don't think it's necessary to make it configurable as all it serves for is to wait for proxy MBean creation which is a side effect of the actual GW sender creation. For this timeout to be reached is an indicator that the MBean may not have been created yet. This may or may not be significant to the user.